File: SampleAppObjC/SampleAppObjC/DevicesDefaultViewController.m
//
//  DevicesDefaultViewController.m
//  WhiteLabelDemoApp
//
//  Created by ani on 7/15/16.
//  Copyright © 2016 LaunchKey. All rights reserved.
//

#import "DevicesDefaultViewController.h"

@interface DevicesDefaultViewController ()
{
    DevicesViewController *devicesChildView;
}
@end

@implementation DevicesDefaultViewController

@synthesize containerView;

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.navigationItem.title =  @"Devices (Default UI)";
    
    devicesChildView = [[DevicesViewController alloc] initWithParentView:self];
    
    [self addChildViewController:devicesChildView];
    [self.containerView addSubview:devicesChildView.view];
    [devicesChildView didMoveToParentViewController:self];
}

-(void)viewWillAppear:(BOOL)animated
{
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(deviceUnlinked) name:deviceUnlinked object:nil];
}

-(void)viewWillDisappear:(BOOL)animated
{
    [[NSNotificationCenter defaultCenter] removeObserver:self name:deviceUnlinked object:nil];
}

-(void)deviceUnlinked
{
    dispatch_async(dispatch_get_main_queue(), ^{
        [self.navigationController popViewControllerAnimated:NO];
    });
}

@end

File: SampleAppSwift/SampleAppSwift/DevicesDefaultViewController.swift
//
//  DevicesDefaultViewController.swift
//  WhiteLabelDemoAppSwift
//
//  Created by ani on 8/7/16.
//  Copyright © 2016 LaunchKey. All rights reserved.
//

import Foundation

class DevicesDefaultViewController:UIViewController
{
    @IBOutlet weak var containerView: UIView!
    
    var devicesChildView:DevicesViewController!
    
    override func viewDidLoad() {
        
        super.viewDidLoad()
        
        self.title = "Devices (Default UI)"
        
        //Navigation Bar Buttons
        self.navigationItem.leftBarButtonItem = UIBarButtonItem(image: UIImage(named: "NavBack"), style: UIBarButtonItem.Style.plain, target: self, action: #selector(ContainerViewController.back))
        self.navigationItem.leftBarButtonItem?.tintColor = UIColor.white
        
        devicesChildView = DevicesViewController.init(parentView: self)
        
        self.addChild(devicesChildView)
        containerView.addSubview(devicesChildView.view)
        devicesChildView.didMove(toParent: self)
        
    }
    
    @IBAction func back()
    {
        if let navController = self.navigationController
        {
            navController.popViewController(animated: true)
        }
    }
}
